theano/sandbox/cuda/cuda_ndarray.cu

```diff
@@ -3244,10 +3244,10 @@ CudaNdarray_gpu_init(PyObject* _unused, PyObject* args)
         if (cnmem > 1)
             mem = cnmem * 1024 * 1024;
         else{
-            // Clip to 98% to let memory for the driver.
-            // 98.5% didn't worked in some cases.
-            if (cnmem > .98){
-                cnmem = .98;
+            // Clip to 95% to let memory for the driver.
+            // 98% didn't worked in some cases.
+            if (cnmem > .99){
```

Comment:
The problem is that there isn't a good way to know the best default. It change by GPUs and other stuff. I think we should have lib.cnmem=1 always work. But we don't want to prevent people to hack there way. Do you have another idea?

I could do cnmem == 1 if you prefer.